### ISSUE TITLE
fix(web): comment_removal_plain missing C++ -> Ccomment grammar swap

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#61]
## Summary

`comment_removal_plain` does not swap C++ to the `Ccomment` grammar the way `comment_removal_json` does, so the two `/comment` endpoints produce different results for `.cpp` / `.cc` / `.h` files.

## Location

- `big-code-analysis-web/src/web/server.rs:60-111`

## Evidence

`comment_removal_json` (lines 60-85) explicitly switches the resolved language to `LANG::Ccomment` when the file is detected as C++:

```rust
let language = if language == LANG::Cpp {
    LANG::Ccomment
} else {
    language
};
HttpResponse::Ok().json(action::<WebCommentCallback>(&language, buf, ...));
```

The sibling endpoint `comment_removal_plain` (lines 87-111) handles the same `/comment` resource for `application/octet-stream` requests but never performs the swap:

```rust
let res = action::<WebCommentCallback>(&language, buf, &PathBuf::from(""), None, cfg);
```

`Ccomment` is a specialized comment-aware grammar (`tree-sitter-ccomment`) used elsewhere in the workspace specifically for C-family comment stripping (see `src/checker.rs:155`, `src/languages/language_ccomment.rs`). Routing C++ through the full C++ grammar instead of `Ccomment` will, at best, produce different output than the JSON variant for the same input; at worst it can fail on inputs the lighter `Ccomment` grammar handles cleanly.

## Expected Behavior

The two `/comment` endpoints should produce identical comment-stripped output for the same input. C++ should be routed through `LANG::Ccomment` in both code paths.

## Actual Behavior

Only `comment_removal_json` performs the C++ → `Ccomment` swap. The plain (`application/octet-stream`) variant uses the full C++ grammar.

## Impact

Inconsistent comment-removal output between content-types for the same input file. Callers that switch transports (e.g., for streaming large files) silently get different results. The discrepancy is also a correctness regression risk: any future fix to the C++ swap logic that updates only one endpoint will widen the divergence.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

The C++ -> `Ccomment` swap was added to `comment_removal_json` but not duplicated into `comment_removal_plain` when the plain (octet-stream) variant was introduced. The two endpoints share the resource path `/comment` and are routed by `content-type` guard, so the divergence is invisible to typical end-to-end tests.

### Implementation Steps

1. Extract the swap into a small helper in `web/server.rs` (or a free function near the top of the module):
   ```rust
   fn comment_language(language: LANG) -> LANG {
       if language == LANG::Cpp { LANG::Ccomment } else { language }
   }
   ```
2. Call the helper from both `comment_removal_json` (line 67-71) and `comment_removal_plain` (line 96).
3. As a quality-of-life fix, consider also routing through `Ccomment` for `LANG::Mozcpp` and any other C-family variants that the comment-removal path supports — verify by searching the library for `rm_comments` callers.

### Tests to Add/Update

- Add a new `#[actix_rt::test]` `test_web_comment_plain_cpp` that posts a `.cpp` file with comments and asserts the result matches what `comment_removal_json` returns for the same input.
- Add a parametrized helper that runs the same source through both endpoints and asserts byte equality, then call it from a single `#[test]` to lock in the parity invariant for future regressions.

### Verification

- [ ] `cargo test -p big-code-analysis-web` passes
- [ ] `cargo clippy -p big-code-analysis-web -- -D warnings` clean
- [ ] Manual verification: `curl -X POST -H "Content-Type: application/octet-stream" --data-binary @sample.cpp 'http://127.0.0.1:8080/comment?file_name=sample.cpp'` returns the same bytes as the JSON variant for the same input.

**Comment by dekobon** [comment]
## Fix Summary
**Root cause**: `comment_removal_plain` never swapped `LANG::Cpp` to `LANG::Ccomment`, unlike its sibling `comment_removal_json`.
**Changes**: `big-code-analysis-web/src/web/server.rs` — extracted a `comment_language()` helper used by both endpoints.
**Tests**: Added `test_web_comment_plain_cpp` — posts a `.cpp` file through the plain endpoint and asserts comments are stripped.
**Commit**: `0b87f711` on branch `fix/issue-61`

**Comment by dekobon** [comment]
## Fix Summary
**Root cause**: comment_removal_plain never swapped C++ to Ccomment grammar
**Changes**: server.rs — extracted comment_language() helper, applied to both endpoints
**Tests**: test_web_comment_plain_cpp
**Commit**: on branch fix/issue-61
